libraries/AP_Scripting/examples/Serial_Dump.lua

```diff
@@ -0,0 +1,44 @@
+-- this script reads data from a serial port and dumps it to a file
+local file_name = 'serial dump.txt'
+local baud_rate = 9600
+
+-- find the serial first (0) scripting serial port instance
+-- SERIALx_PROTOCOL 28
+local port = assert(serial:find_serial(0),"Could not find Scripting Serial Port")
+
+-- make a file
+local file = assert(io.open(file_name, "w"),"Could not create file " .. file_name)
+file:close()
+
+-- begin the serial port
+port:begin(baud_rate)
+port:set_flow_control(0)
+
+-- table to buffer data
+local buffer = {}
```

Comment:
Just make this local to inside the while loop in update (line 24), since it's thrown out every time keeping it as an upvalue doesn't help, and would allow you to remove setting it on line 37

Reply:
done